### PR TITLE
Validation translations

### PR DESCRIPTION
**Pull request by pieterbos**
- Better and translatable validation messages for archetype validation
- Dutch translations for these validation messages

### CODE REVIEW

File: tools/src/main/java/com/nedap/archie/archetypevalidator/ErrorType.java [hunk 1]
```diff
@@ -1,84 +1,87 @@
...
+    WACMCL(I18n.register("cardinality/occurrences lower bound validity: where a cardinality with a finite upper bound is stated on an attribute, for all immediate child objects for which an occurrences constraint is stated, the sum of occurrences lower bounds should be lower than the cardinality upper limit.")),
+    VDSEV(I18n.register("archetype slot 'exclude' constraint validity. The 'exclude' constraint in an archetype slot must conform to the slot constraint validity rules.")),
+    VACSO(I18n.register("single-valued attribute child object occurrences validity: the occurrences of a child object of a single-valued attribute cannot have an upper limit greater than 1.")),
+    VACMCU(I18n.register(" cardinality/occurrences upper bound validity: where a cardinality with a finite upper bound is stated on an attribute, for all immediate child objects for which an occurrences constraint is stated, the occurrences must either have an open upper bound (i.e. n..*) which is interpreted as the maximum value allowed within the cardinality, or else a finite upper bound which is ⇐ the cardinality upper bound.")),
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/ErrorType.java]:
```suggestion
    VACMCU(I18n.register("cardinality/occurrences upper bound validity: where a cardinality with a finite upper bound is stated on an attribute, for all immediate child objects for which an occurrences constraint is stated, the occurrences must either have an open upper bound (i.e. n..*) which is interpreted as the maximum value allowed within the cardinality, or else a finite upper bound which is ⇐ the cardinality upper bound.")),
```

File: i18n/po/i18n_nl.po [hunk 2]
```diff
@@ -391,6 +808,44 @@ msgstr "minimaal {0}"
 msgid "at most {0}"
 msgstr "maximaal {0}"
 
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/ValidationMessage.java:73
+msgid "at path: "
+msgstr "o\\p het pad: "
```
Comment (J3173) [inline i18n/po/i18n_nl.po]: Why `o\\p`?

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/ValidateAgainstReferenceModel.java [hunk 3]
```diff
@@ -88,40 +93,53 @@ public void validate(CAttribute cAttribute) {
...
+                                            I18n.t("The existence of attribute {0}.{1} is the same as in the reference model - this is not allowed due to strict existence validation being enabled",
+                                                    owningObject.getRmTypeName(), cAttribute.getRmAttributeName()));
                                 } else {
                                     //TODO: warn
                                     cAttribute.setExistence(null);
                                 }
                             } else {
-                                addMessageWithPath(ErrorType.VCAEX, cAttribute.path());
+                                addMessageWithPath(ErrorType.VCAEX, cAttribute.path(),
+                                        I18n.t("existence {0} of attribute {2}.{3} does not match existence {1} of the reference model",
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/ValidateAgainstReferenceModel.java]:
```suggestion
                                        I18n.t("The existence {0} of attribute {2}.{3} does not match existence {1} of the reference model",
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/ValidateAgainstReferenceModel.java [hunk 4]
```diff
@@ -88,40 +93,53 @@ public void validate(CAttribute cAttribute) {
...
+                                                    owningObject.getRmTypeName(), cAttribute.getRmAttributeName()));
                                 } else {
-                                    //TODO: warning
+                                    //TODO: warning. Or just don't serialize?
                                     cAttribute.setCardinality(null);
                                 }
                             } else {
-                                addMessageWithPath(ErrorType.VCACA, cAttribute.path());
+                                addMessageWithPath(ErrorType.VCACA, cAttribute.path(),
+                                        I18n.t("cardinality {0} of attribute {2}.{3} does not match cardinality {1} of the reference model",
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/ValidateAgainstReferenceModel.java]:
```suggestion
                                        I18n.t("The cardinality {0} of attribute {2}.{3} does not match cardinality {1} of the reference model",
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/CodeValidation.java [hunk 5]
```diff
@@ -27,11 +29,14 @@ public void validate(CObject cObject) {
         int codeSpecializationDepth = AOMUtils.getSpecializationDepthFromCode(nodeId);
         int archetypeSpecializationDepth = archetype.specializationDepth();
         if(codeSpecializationDepth > archetypeSpecializationDepth) {
-            addMessageWithPath(ErrorType.VTSD, cObject.path());
+            addMessageWithPath(ErrorType.VTSD, cObject.path(),
+                    I18n.t("code specialization depth of code {0} is {1}, which is greater than archetype specialization depth {2}",
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/CodeValidation.java]:
```suggestion
                    I18n.t("The code specialization depth of code {0} is {1}, which is greater than archetype specialization depth {2}",
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/CodeValidation.java [hunk 6]
```diff
@@ -27,11 +29,14 @@ public void validate(CObject cObject) {
...
+                            nodeId, codeSpecializationDepth, archetypeSpecializationDepth));
         } else if (cObject.isRoot() || parentIsMultiple(cObject)) {
             if( (codeSpecializationDepth < archetypeSpecializationDepth && !flatParent.getTerminology().hasIdCode(nodeId)) ||
                     (codeSpecializationDepth == archetypeSpecializationDepth &&  !archetype.getTerminology().hasIdCode(nodeId))) {
-                addMessageWithPath(ErrorType.VATID, cObject.path(), nodeId + " is not defined in terminology or parent terminology");
+                addMessageWithPath(ErrorType.VATID, cObject.path(),
+                        I18n.t("node id {0} is used in the archetype, but missing in the terminology", nodeId));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/CodeValidation.java]:
```suggestion
                        I18n.t("Node id {0} is used in the archetype, but missing in the terminology", nodeId));
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/NodeIdValidation.java [hunk 7]
```diff
@@ -30,11 +31,12 @@ protected void beginValidation() {
...
         if(cObject.getNodeId() == null) {
             //every object must have a node id
-            addMessageWithPath(ErrorType.VCOID, cObject.getPath());
+            addMessageWithPath(ErrorType.VCOID, cObject.getPath(),
+                    I18n.t("C_Object with RM type {0} must have a node id", cObject.getRmTypeName()));
         }
         else if(!CPrimitiveObject.PRIMITIVE_NODE_ID_VALUE.equals(cObject.getNodeId()) && nodeIds.containsKey(cObject.getNodeId())) {
             //every node id in a single archetype must be unique or a primitive object
-            addMessageWithPath(ErrorType.VCOSU, cObject.getPath(), "node ID " + cObject.getNodeId() + " already used in " + nodeIds.get(cObject.getNodeId()));
+            addMessageWithPath(ErrorType.VCOSU, cObject.getPath(), I18n.t("node ID {0} already used in path {1}", cObject.getNodeId(), nodeIds.get(cObject.getNodeId())));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/NodeIdValidation.java]:
```suggestion
            addMessageWithPath(ErrorType.VCOSU, cObject.getPath(), I18n.t("Node id {0} already used in path {1}", cObject.getNodeId(), nodeIds.get(cObject.getNodeId())));
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/AuthoredArchetypeMetadataChecks.java [hunk 8]
```diff
@@ -45,22 +46,22 @@ private void checkLanguagesInTranslationAreInTerminology() {
             for(String language:archetype.getTranslations().keySet()) {
                 TranslationDetails translationDetails = archetype.getTranslations().get(language);
                 if(translationDetails.getLanguage() == null || !language.equals(translationDetails.getLanguage().getCodeString())) {
-                    addMessage(ErrorType.VTRLA, String.format("key for language % is wrong: %s", language, translationDetails.getLanguage().getCodeString()));
+                    addMessage(ErrorType.VTRLA, I18n.t("Translation details language {0} has an incorrect key: {1}", language, translationDetails.getLanguage().getCodeString()));
                 }
                 //check if also defined in terminology
                 if(archetype.getTerminology().getTermDefinitions().get(language) == null) {
-                    addMessage(ErrorType.VOTM, String.format("language %s defined in translations, but not defined in terminology", language));
+                    addMessage(ErrorType.VOTM, I18n.t("language {0} is defined in the translations, but is not present in the terminology", language));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/AuthoredArchetypeMetadataChecks.java]:
```suggestion
                    addMessage(ErrorType.VOTM, I18n.t("Language {0} is defined in the translations, but is not present in the terminology", language));
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/AuthoredArchetypeMetadataChecks.java [hunk 9]
```diff
@@ -69,7 +70,7 @@ private void checkOriginalLanguagePresent() {
             String languageCode = archetype.getOriginalLanguage().getCodeString();;
             if(languageCode != null) {
                 if(archetype.getTerminology().getTermDefinitions().get(languageCode) == null) {
-                    addMessage(ErrorType.VOLT, String.format("original language %s not defined in terminology", languageCode));
+                    addMessage(ErrorType.VOLT, I18n.t("original language {0} is not defined in the terminology", languageCode));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/AuthoredArchetypeMetadataChecks.java]:
```suggestion
                    addMessage(ErrorType.VOLT, I18n.t("Original language {0} is not defined in the terminology", languageCode));
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/ValidateAgainstReferenceModel.java [hunk 10]
```diff
@@ -88,40 +93,53 @@ public void validate(CAttribute cAttribute) {
...
+                                                cAttribute.getCardinality(), defaultAttribute.getCardinality(),
+                                                owningObject.getRmTypeName(), cAttribute.getRmAttributeName()));
                             }
                         }
                     } else {
                         if(cAttribute.getCardinality() != null) {
-                            addMessageWithPath(ErrorType.VCAM, defaultAttribute.path(), "single valued attributes can not have a cardinality");
+                            addMessageWithPath(ErrorType.VCAM, defaultAttribute.path(),
+                                    I18n.t("single valued attributes can not have a cardinality"));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/ValidateAgainstReferenceModel.java]:
```suggestion
                                    I18n.t("Single valued attributes can not have a cardinality"));
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/ArchetypeValidator.java [hunk 11]
```diff
@@ -184,7 +185,7 @@ public ValidationResult validate(Archetype archetype, FullArchetypeRepository re
             result.addOverlayValidations(extraArchetypeRepository.getAllValidationResults());
             for(ValidationResult subResult:extraArchetypeRepository.getAllValidationResults()) {
                 if(!subResult.passes()) {
-                    result.getErrors().add(new ValidationMessage(ErrorType.OTHER, MessageFormat.format("template overlay {0} had validation errors", subResult.getArchetypeId())));
+                    result.getErrors().add(new ValidationMessage(ErrorType.OTHER, I18n.t("template overlay {0} had validation errors", subResult.getArchetypeId())));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/ArchetypeValidator.java]:
```suggestion
                    result.getErrors().add(new ValidationMessage(ErrorType.OTHER, I18n.t("Template overlay {0} had validation errors", subResult.getArchetypeId())));
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/BasicChecks.java [hunk 12]
```diff
@@ -37,19 +38,19 @@ private void checkSpecializationDepth() {
...
 
     private void checkIdCodeSpecialisationLevel() {
         int depth = ValidationUtils.getSpecializationDepth(archetype, repository);
         if(depth != archetype.getDefinition().specialisationDepth()) {
-            addMessage(ErrorType.VARCN);
+            addMessage(ErrorType.VARCN, I18n.t("Incorrect root node id {0}: it must match the specialization depth of the archetype, which is {1}", archetype.getDefinition().getNodeId(), depth));
         }
         if(!archetype.getDefinition().getNodeId().matches("id1(.1)*")) {
-            addMessage(ErrorType.VARCN, archetype.getDefinition().getNodeId());
+            addMessage(ErrorType.VARCN, I18n.t("the node id is not in the form id1.1....1: {0}", archetype.getDefinition().getNodeId()));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/BasicChecks.java]:
```suggestion
            addMessage(ErrorType.VARCN, I18n.t("The node id is not in the form id1.1....1: {0}", archetype.getDefinition().getNodeId()));
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/NodeIdValidation.java [hunk 13]
```diff
@@ -30,11 +31,12 @@ protected void beginValidation() {
     public void validate(CObject cObject) {
         if(cObject.getNodeId() == null) {
             //every object must have a node id
-            addMessageWithPath(ErrorType.VCOID, cObject.getPath());
+            addMessageWithPath(ErrorType.VCOID, cObject.getPath(),
+                    I18n.t("C_Object with RM type {0} must have a node id", cObject.getRmTypeName()));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/NodeIdValidation.java]:
```suggestion
                    I18n.t("C_OBJECT with RM type {0} must have a node id", cObject.getRmTypeName()));
```

File: i18n/po/i18n_nl.po [hunk 14]
```diff
@@ -2,45 +2,189 @@ msgid ""
...
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java:94
+#, java-format
+msgid "A {0} cannot specialize a {1}"
+msgstr "Een {0} kan niet een {1} specialiseren"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/AuthoredArchetypeMetadataChecks.java:61
+#, java-format
+msgid "ADL version {0} is an invalid format for a version, should be x.x.x-(rc|alpha(x)?)?"
+msgstr "ADL versie {0} is in een incorrect formaat. Dit zou een versie in het formaat x.x.x-(rc|alpha(x)?)? moeten zijn"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "ADL-versie {0} is in een incorrect formaat. Dit zou een versie in het formaat x.x.x-(rc|alpha(x)?)? moeten zijn"
```

File: i18n/po/i18n_nl.po [hunk 15]
```diff
@@ -2,45 +2,189 @@ msgid ""
...
+msgstr "Attribuut {0}.{1} mag geen type {2}  bevatten"
+
 #: ../bmm/src/main/java/org/openehr/bmm/persistence/validation/BmmMessageIds.java:92
 msgid "BMM Schema load error"
 msgstr "Fout bij het laden van een BMM schema"
 
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java:249
+#, java-format
+msgid "C_OBJECT in this archetype with class {0} is prohibited, which means its class must be the same as parent type {1}"
+msgstr "Het C_OBJECT  in dit archetype met klasse {0} heeft 0 als maximum occurrences (prohibited), maar heeft niet dezelfde klasse als het C_OBJECT dat gespecializeerd wordt: {1}"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "Het C_OBJECT in dit archetype met klasse {0} heeft 0 als maximum occurrences (prohibited), maar heeft niet dezelfde klasse als het C_OBJECT dat gespecializeerd wordt: {1}"
```

File: i18n/po/i18n_nl.po [hunk 16]
```diff
@@ -2,45 +2,189 @@ msgid ""
...
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/CodeValidation.java:76
+#, java-format
+msgid "Code {0} from the C_TERMINOLOGY_CODE constraint is not defined in the terminology"
+msgstr "Code {0} van deze C_TERMINOLOGY_CODE bestaat niet in de terminology"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java:47
+#, java-format
+msgid "Could not find parent object for {0} but it should have been prechecked. Could you report this as a bug?"
+msgstr "Kan parent object voor {0} niet vinden. Deze melding is waarschijnlijk een bug, zou U dat kunnen melden?"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "Kan parent object voor {0} niet vinden. Deze melding is waarschijnlijk een bug, zou u dat kunnen melden?"
```

File: i18n/po/i18n_nl.po [hunk 17]
```diff
@@ -86,7 +260,27 @@ msgstr ""
 #: ../bmm/src/main/java/org/openehr/bmm/persistence/validation/BmmMessageIds.java:37
 #, java-format
 msgid "No Reference Model schema found for package ''{0}''"
-msgstr ""
+msgstr "Geen reference model schema gevonden voor package “{0}”"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "Geen referentiemodel-schema gevonden voor package “{0}”"
```

File: i18n/po/i18n_nl.po [hunk 18]
```diff
@@ -98,10 +292,34 @@ msgstr "Object komt niet overeen met tupel: {0}"
...
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java:122
+#, java-format
+msgid "Primitive object with rm type {0} does not conform to primitive object with rm type {1} in parent"
+msgstr "Primitive object met rm type {0} klopt niet bij primitive object met rm type {1} in het archetype dat gespecialiseerd wordt"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/AuthoredArchetypeMetadataChecks.java:64
+#, java-format
+msgid "RM Release version {0} is an invalid format for a version, should be x.x.x-(rc|alpha(x)?)?"
+msgstr "Het formaat van de rm release version {0} klopt niet. dit moet in het formaat  x.x.x-(rc|alpha(x)?)? zijn"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "Het formaat van de RM release version {0} klopt niet. dit moet in het formaat  x.x.x-(rc|alpha(x)?)? zijn"
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java [hunk 19]
```diff
@@ -99,15 +109,22 @@ private void validateConformsTo(CObject cObject, CObject parentCObject) {
...
+                addMessageWithPath(ErrorType.VSONCO, cObject.path(),
+                        I18n.t("Occurrences {0} does not conform to occurrences {1} in parent", cObject.getOccurrences(), parentCObject.getOccurrences()));
             } else if (!cObject.nodeIdConformsTo(parentCObject)) {
-                addMessageWithPath(ErrorType.VSONI, cObject.path());
+                addMessageWithPath(ErrorType.VSONI, cObject.path(),
+                        I18n.t("Node ID {0} does not conform to node id {1} in parent", cObject.getNodeId(), parentCObject.getNodeId()));
             } else if (cObject instanceof CPrimitiveObject && parentCObject instanceof CPrimitiveObject) {
-                addMessageWithPath(ErrorType.VPOV, cObject.path());
+                addMessageWithPath(ErrorType.VPOV, cObject.path(),
+                        I18n.t("Primitive object with rm type {0} does not conform to primitive object with rm type {1} in parent",
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java]:
```suggestion
                        I18n.t("Primitive object with RM type {0} does not conform to primitive object with RM type {1} in parent",
```

File: i18n/po/i18n_nl.po [hunk 20]
```diff
@@ -2,45 +2,189 @@ msgid ""
...
+msgid "Archetype referenced in use_archetype points to class {0}, which does not exist in this reference model"
+msgstr "Het archetype waarnaar use_archetype verwijst, verwijst naar klasse {0}. Deze klasse bestaat niet in het referentiemodel."
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/VariousStructureValidation.java:74
+msgid "Archetype root must have an archetype reference or be prohibited (occurrences matches 0)"
+msgstr "Use_archetype moet verwijzen naar een archetype, of beperkt zijn tot occurrences 0"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/ErrorType.java:39
+msgid "Archetype root must reference an existing archetype"
+msgstr "Use_archetype moet verwijzen naar een bestaand archetype"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "use_archetype moet verwijzen naar een bestaand archetype"
```

File: i18n/po/i18n_nl.po [hunk 21]
```diff
@@ -202,6 +420,11 @@ msgstr ""
 msgid "Reference model schema(s) {0} specified in options not valid or not found in schema directories"
 msgstr ""
 
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/AuthoredArchetypeMetadataChecks.java:37
+#, java-format
+msgid "Resource description language {0} has an incorrect key: {1}"
+msgstr ""
```
Comment (J3173) [inline i18n/po/i18n_nl.po]: Translation missing

File: i18n/po/i18n_nl.po [hunk 22]
```diff
@@ -297,66 +515,179 @@ msgstr ""
...
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/MultiplicitiesValidation.java:34
+#, java-format
+msgid "The attribute {0} of type {1} can only have a single value, but the occurrences of the C_OBJECTS below has an upper limit of more than 1"
+msgstr "Het attribuut {0} van type {1} kan maar één waarde hebben, maar de occurrences van de C_OBJECTS in dit attibuut heeft een maximale limiet van meer dan 1"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/ValidateAgainstReferenceModel.java:126
+#, java-format
+msgid "The cardinality of Attribute {0}.{1} is the same as in the reference model - this is not allowed due to strict multiplicities validation being enabled"
+msgstr "De cardinaliteit van attirbuut {0}.{1} is hetzelfde als in het referentie model. Dit is niet toegestaan als strikte validatie is ingeschakeld"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "De cardinaliteit van attribuut {0}.{1} is hetzelfde als in het referentiemodel. Dit is niet toegestaan als strikte validatie is ingeschakeld"
```

File: i18n/po/i18n_nl.po [hunk 23]
```diff
@@ -297,66 +515,179 @@ msgstr ""
...
+msgstr "De specialisatie depth van het archetype, {0}, moet één groter zijn dan die van het gespecialiseerde archetype, {1}"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/ErrorType.java:28
+msgid "The specialisation depth of the archetypes must be one greater than the specialisation depth of the parent archetype"
+msgstr "De specialisatie depth van het archetype moet één groter zijn dan die van het gespecialiseerde archetype"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java:192
+#, java-format
+msgid "The use_archetype with type {0} does not match the archetype slow (allow_archetype) with type {1}"
+msgstr "Use_archetype met {0} klopt niet met het archetype slot met type {1} uit het gespecialiseerde archetype"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "use_archetype met {0} klopt niet met het archetype slot met type {1} uit het gespecialiseerde archetype"
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java [hunk 24]
```diff
@@ -20,11 +21,11 @@ protected void validate(CComplexObjectProxy cObject) {
         //validate that CComplexObjectProxy nodes have a valid path
         ComplexObjectProxyReplacement complexObjectProxyReplacement = ComplexObjectProxyReplacement.getComplexObjectProxyReplacement(cObject);
         if(complexObjectProxyReplacement == null) {
-            addMessageWithPath(ErrorType.VUNP, cObject.path());
+            addMessageWithPath(ErrorType.VUNP, cObject.path(), I18n.t("Use node (C_COMPLEX_OBJECT_PROXY) points to a path that cannot be found: {0}", cObject.getTargetPath()));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java]:
```suggestion
            addMessageWithPath(ErrorType.VUNP, cObject.path(), I18n.t("use_node (C_COMPLEX_OBJECT_PROXY) points to a path that cannot be found: {0}", cObject.getTargetPath()));
```

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java [hunk 25]
```diff
@@ -20,11 +21,11 @@ protected void validate(CComplexObjectProxy cObject) {
...
         } else {
             CComplexObject replacement = complexObjectProxyReplacement.getReplacement();
             if(!combinedModels.rmTypesConformant(replacement.getRmTypeName(), cObject.getRmTypeName())) {
-                addMessageWithPath(ErrorType.VUNT, cObject.path());
+                addMessageWithPath(ErrorType.VUNT, cObject.path(), I18n.t("Use node (C_COMPLEX_OBJECT_PROXY) points to type {0}, which does not conform to type {1}", replacement.getRmTypeName(), cObject.getRmTypeName()));
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/FlatFormValidation.java]:
```suggestion
                addMessageWithPath(ErrorType.VUNT, cObject.path(), I18n.t("use_node (C_COMPLEX_OBJECT_PROXY) points to type {0}, which does not conform to type {1}", replacement.getRmTypeName(), cObject.getRmTypeName()));
```

File: i18n/po/i18n_nl.po [hunk 26]
```diff
@@ -297,66 +515,179 @@ msgstr ""
...
+msgstr "Het pad {0} gebruikt in de annotations bestaat niet in het flattened archetype of referentiemodel"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/ErrorType.java:30
+msgid "The rm_release top-level meta-data item must exist and consist of a valid 3-part version identifier."
+msgstr ""
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/BasicChecks.java:32
+#, java-format
+msgid "The specialisation depth of the archetype, {0}, must be one greater than the specialisation depth of the parent archetype, {1}"
+msgstr "De specialisatie depth van het archetype, {0}, moet één groter zijn dan die van het gespecialiseerde archetype, {1}"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "De specialisation depth van het archetype, {0}, moet één groter zijn dan die van het gespecialiseerde archetype, {1}"
```

File: i18n/po/i18n_nl.po [hunk 27]
```diff
@@ -297,66 +515,179 @@ msgstr ""
...
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/ErrorType.java:28
+msgid "The specialisation depth of the archetypes must be one greater than the specialisation depth of the parent archetype"
+msgstr "De specialisatie depth van het archetype moet één groter zijn dan die van het gespecialiseerde archetype"
```
Comment (J3173) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "De specialisation depth van het archetype moet één groter zijn dan die van het gespecialiseerde archetype"
```

File: i18n/po/i18n_nl.po (hunk 2 above)
Comment (pieterbos) [reply i18n/po/i18n_nl.po]:
```suggestion
msgstr "op het pad: "
```

File: i18n/po/i18n_nl.po (hunk 21 above)
Comment (pieterbos) [reply i18n/po/i18n_nl.po]: This is a very technical sentence, that can only be found by editing the object model or introducing a bug - not by editing ADL. Not translating at the moment.

File: i18n/po/i18n_nl.po [hunk 28]
```diff
@@ -400,17 +855,254 @@ msgstr "leeg"
...
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java:166
+#, java-format
+msgid "use_archetype specializes an archetype root pointing to {0}, but the archetype {1} is not a descendant"
+msgstr "use_archetype {1} specialiseert een andere use_archetype die verwijst naar {0}. Maar het archeytpe {1} specialiseert niet {0}"
+
+#: ../tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java:207
+#, java-format
+msgid "use_archetype {0} does not match the expression of the archetype slot it specialized in the parent"
+msgstr "use_archetype {0} komt niet overeen met de restricties van het archetype slot dat het specialiseert"
```
Comment (pieterbos) [inline i18n/po/i18n_nl.po]:
```suggestion
msgstr "Use_archetype {0} komt niet overeen met de restricties van het archetype slot dat het specialiseert"
```

**Pull request merged**